### PR TITLE
azure-cli-extensions.application-insights: init at 1.2.1

### PR DESCRIPTION
**Pull request by andreasuvoss**
## Description of changes

Extension for the azure-cli. 
Adds commands for working with Azure Application Insights.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. 
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by andreasuvoss** [comment]
@katexochen would you mind reviewing this?

**Comment by katexochen** [comment]
Please don't create merge commits but rather rebase our PR on latest master.

File: pkgs/by-name/az/azure-cli/extensions-manual.nix [hunk 1]
```diff
@@ -38,6 +38,18 @@
     meta.maintainers = with lib.maintainers; [ obreitwi ];
   };
 
+  application-insights = mkAzExtension rec {
```
Comment (katexochen) [inline pkgs/by-name/az/azure-cli/extensions-manual.nix:41]: Please try to keep the extensions in alphabetical order.

**Review by katexochen — CHANGES_REQUESTED** [review]
Otherwise looks good to me.

Were you able to build the cli with the extension and tests its functionality?

File: pkgs/by-name/az/azure-cli/extensions-manual.nix (hunk 1 above)
Comment (andreasuvoss) [reply pkgs/by-name/az/azure-cli/extensions-manual.nix]: Done!

**Comment by andreasuvoss** [comment]
> Otherwise looks good to me.
> 
> Were you able to build the cli with the extension and tests its functionality?

Yes, I was able to build it locally and run the CLI with the extension from the `result/bin/az` binary.

**Review by katexochen — APPROVED** [review]
Result of `nixpkgs-review pr 326386` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>azure-cli-extensions.application-insights</li>
    <li>azure-cli-extensions.application-insights.dist</li>
  </ul>
</details>

**Comment by katexochen** [comment]
Thanks, and welcome as nixpkgs maintainer! :)

**Pull request merged**